src/SFA.DAS.Tools.Servicebus.Support.Functions/CleanExpiredUserSessionsFunction.cs

```diff
@@ -0,0 +1,87 @@
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.Tools.Servicebus.Support.Application;
+using SFA.DAS.Tools.Servicebus.Support.Application.Queue.Commands.DeleteUserSession;
+using SFA.DAS.Tools.Servicebus.Support.Application.Queue.Queries.GetExpiredUserSessions;
+using SFA.DAS.Tools.Servicebus.Support.Application.Queue.Queries.GetMessages;
+using SFA.DAS.Tools.Servicebus.Support.Application.Services;
+using SFA.DAS.Tools.Servicebus.Support.Infrastructure.Services;
+using System;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.Tools.Servicebus.Support.Functions
+{
+    public class CleanExpiredUserSessionsFunction
+    {
+        private readonly IQueryHandler<GetExpiredUserSessionsQuery, GetExpiredUserSessionsQueryResponse> _expiredUserSessionQuery;
+        private readonly IQueryHandler<GetMessagesQuery, GetMessagesQueryResponse> _getMessagesQuery;
+        private readonly IMessageService _messageService;
+        private readonly ICommandHandler<DeleteUserSessionCommand, DeleteUserSessionCommandResponse> _deleteUserSessionCommand;
+        private readonly IUserService _userService;
+
+        public CleanExpiredUserSessionsFunction(
+            IQueryHandler<GetExpiredUserSessionsQuery, GetExpiredUserSessionsQueryResponse> expiredUserSessionQuery,
+            IQueryHandler<GetMessagesQuery, GetMessagesQueryResponse> getMessagesQuery,
+            IMessageService messageService,
+            ICommandHandler<DeleteUserSessionCommand, DeleteUserSessionCommandResponse> deleteUserSessionCommand,
+            IUserService userService
+            )
+        {
+            _expiredUserSessionQuery = expiredUserSessionQuery;
+            _getMessagesQuery = getMessagesQuery;
+            _messageService = messageService;
+            _deleteUserSessionCommand = deleteUserSessionCommand;
+            _userService = userService;
+        }
+
+        [FunctionName("CleanExpiredUserSessionsFunction")]
+        public async Task Run([TimerTrigger("%CleanExpiredUserSessionsFunctionTimer%")] TimerInfo myTimer, ILogger log)
+        {
+            try
+            {
+                var queryResult = await _expiredUserSessionQuery.Handle(new GetExpiredUserSessionsQuery());
+
+                foreach (var session in queryResult.ExpiredUserSessions)
+                {
+                    _userService.Configure(session.UserId, "CleanExpiredUserSessionsFunction");
+
+                    var getMessagesResponse = await GetMessages(session.UserId);
+
+                    while (getMessagesResponse?.Count > 0)
+                    {
+                        await _messageService.AbortMessages(getMessagesResponse.Messages, session.Queue);
+                        getMessagesResponse = await GetMessages(session.UserId);
+                    }
+
+                    if (getMessagesResponse?.Count == 0)
```

Comment:
Should you be using .Any() here rather than checking the count?

Reply:
@DannyGB modified